modules/table/src/test/java/org/apache/ignite/internal/table/distributed/replication/PartitionReplicaListenerTest.java

```diff
@@ -704,6 +705,30 @@ public void testTxStateReplicaRequestCommitState() throws Exception {
         assertTrue(readTimestamp.compareTo(txMeta.commitTimestamp()) > 0);
     }
 
+    @Test
+    public void testExecuteRequestOnFinishedTx() {
```

Comment:
- I'd rather convert it to parametrized test with current state as a parameter [COMMITTED, ABORTED, FIINISHING].
- I'd also iterate through all possible RequestType options.